### PR TITLE
[Listener] Check freshness of data pulled from blockchain

### PR DESCRIPTION
**Pull request by franckc**
First pull request? Read our [guide to contributing](http://docs.originprotocol.com/#contributing)

### Description:

Add extra check on listing and offer data read from the blockchain to avoid scenario where stale data is pulled from an out of sync node then persisted to DB and ElasticSearch.
This is defensive coding - I haven't noticed this yet in production but I'm fairly sure it would happen sooner or later...

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### CODE REVIEW

File: origin-discovery/test/listener-rules.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  it(`Should throw`, () => {
+    const events = []
+    const blockInfo = {
+      blockNumber: 1,
+      logIndex: 1
+    }
+    expect(() => checkFreshness(events, blockInfo)).to.throw()
+  })
+})
```
Comment (sparrowDom) [inline origin-discovery/test/listener-rules.test.js]: nit: new line at the end of file

File: origin-discovery/src/listener/rules.js [hunk 2]
```diff
@@ -20,6 +20,28 @@ function generateOfferId(log) {
...
+  let fresh = false
+  events.forEach(event => {
+    if ((event.blockNumber > blockInfo.blockNumber) ||
+      (event.blockNumber === blockInfo.blockNumber && event.logIndex >= blockInfo.logIndex)) {
+      fresh = true
+    }
+  })
+  if (!fresh) {
+    throw new Error('Freshness check failed')
+  }
```
Comment (sparrowDom) [inline origin-discovery/src/listener/rules.js]: a shorter alternative: 
```
// Find at least 1 event that is as fresh as blockInfo.
if (!events.some(event => {
  return (event.blockNumber > blockInfo.blockNumber) ||
    (event.blockNumber === blockInfo.blockNumber && event.logIndex >= blockInfo.logIndex)
})) {
  throw new Error('Freshness check failed')
}
```

Comment (franckc) [reply origin-discovery/src/listener/rules.js]: oh I did not know about some method on arrays. very useful. implemented. thank you ! :)

File: origin-discovery/test/listener-rules.test.js (hunk 1 above)
Comment (franckc) [reply origin-discovery/test/listener-rules.test.js]: done

**Pull request merged**